compiler/next/test/resolution/testPoi.cpp

```diff
@@ -0,0 +1,604 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/parsing/parsing-queries.h"
+#include "chpl/resolution/resolution-queries.h"
+#include "chpl/resolution/scope-queries.h"
+#include "chpl/uast/Call.h"
+#include "chpl/uast/Comment.h"
+#include "chpl/uast/Identifier.h"
+#include "chpl/uast/Module.h"
+#include "chpl/uast/Variable.h"
+
+// always check assertions in this test
+#ifdef NDEBUG
+#undef NDEBUG
+#endif
+
+#include <cassert>
+
+using namespace chpl;
+using namespace parsing;
+using namespace resolution;
+using namespace uast;
+
+// test showing point-of-instantiation behavior
+static void test1() {
+  printf("test1\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  auto path = UniqueString::build(context, "input.chpl");
+  std::string contents = "module M {\n"
+                         "  proc generic(param p) {\n"
+                         "    helper();\n"
+                         "  }\n"
+                         "}\n"
+                         "module N {\n"
+                         "  use M;\n"
+                         "  proc helper() { }\n"
+                         "  generic(1);\n"
+                         "}\n";
+  setFileText(context, path, contents);
+
+  const ModuleVec& vec = parse(context, path);
+  assert(vec.size() == 2);
+  const Module* m = vec[0]->toModule();
+  assert(m);
+  assert(m->numStmts() == 1);
+  const Module* n = vec[1]->toModule();
+  assert(n);
+  assert(n->numStmts() == 3);
+  const Function* mGeneric = m->stmt(0)->toFunction();
+  assert(mGeneric);
+  const Call* helperCall = mGeneric->stmt(0)->toCall();
+  assert(helperCall);
+  const Function* nHelper = n->stmt(1)->toFunction();
+  assert(nHelper);
+  const Call* genericCall = n->stmt(2)->toCall();
+  assert(genericCall);
+
+  // resolve module N
+  const ResolutionResultByPostorderID& rr = resolveModule(context, n->id());
+
+  // get the resolved function
+  const ResolvedFunction* rfunc =
+    resolveOnlyCandidate(context, rr.byAst(genericCall));
+  assert(rfunc);
```

Comment:
Would assert(rfunc->functionId() == mGeneric->id()) evaluate true in this case. 

Or is it that since this is a call to a generic function we make a new specialization function and rfunc->functionId() would be something else? If that's the case maybe add in assert(rfunc->functionId() != mGeneric->id()) (with a comment) to demonstrate that.

Is there anything we could add to lock down that this is resolving to a specialization of generic?

Reply:
> Would assert(rfunc->functionId() == mGeneric->id()) evaluate true in this case.

Yes

> Is there anything we could add to lock down that this is resolving to a specialization of generic?

Yes, I've added a check that the result is an instantiated generic.